### PR TITLE
[4.0] Fields/editor/none as custom element

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
Textarea (AKA editor none) implementing the Joomla API for editors 

### Testing Instructions
Install from this PR's branch. Change the default editor to none. Check that the buttons work and they do:
- insert at the start of the text
- insert at the end of the text
- after making some selection and trying some of the buttons (the value from the xtd-button should replace the texture selection).

Mind that modals are not closing, it's a known issue!

### CODE REVIEW

**Comment by samarthsharma36** [comment]
@dgt41  Are you talking about this editor 
And if you are, what exactly are you trying to test? Closing the Modal?
Please help me test your patch

selection_025

**Comment by dgrammatiko** [comment]
@Samgithub1997 nope, that's the default editor (tinyMCE) you need to go to Global Configuration and change the default editor to none:
screen shot 2018-03-19 at 21 07 07

Then try to insert, eg an image using the buttons bellow the editor...

**Comment by samarthsharma36** [comment]
@dgt41  Thanks a lot!

**Comment by Samgithub1997** [comment]
I tried the your patch but I am not able to understand the difference @dgt41

**Comment by dgrammatiko** [comment]
Nice, so if there is no difference (there shouldn't be) then this is a passing test. All that is happening with this PR is moving the javascript form the old version (eg DOMConentLoaded) to the brand new web components. There is no other enhancement here...

**Comment by samarthsharma36** [comment]
Alright ! Thanks

**Comment by anuragteapot** [comment]
@Samgithub1997 Thanks 
Now go  https://issues.joomla.org/tracker/joomla-cms/19909  and mark test result.

**Comment by Samgithub1997** [comment]
I have tested this item :white_check_mark: successfully on 0adcf61ca5835d8109b434dd85855619b608eda7<br /><br />The test was successful. It wasn't an issue but basically an up-gradation of PR from older JS version to a new web component.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19909">issues.joomla.org/tracker/joomla-cms/19909</a>.</sub>

**Comment by samarthsharma36** [comment]
I have tested this item :white_check_mark: successfully on 0adcf61ca5835d8109b434dd85855619b608eda7<br /><br />The test was successful. It wasn't an issue but basically a necessary up-gradation from older JS version to a new web component.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19909">issues.joomla.org/tracker/joomla-cms/19909</a>.</sub>

**Comment by anuragteapot** [comment]
I have tested this item :white_check_mark: successfully on 0adcf61ca5835d8109b434dd85855619b608eda7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19909">issues.joomla.org/tracker/joomla-cms/19909</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19909">issues.joomla.org/tracker/joomla-cms/19909</a>.</sub>

File: plugins/editors/none/none.php [hunk 1]
```diff
@@ -72,11 +59,16 @@ public function onDisplay($name, $content, $width, $height, $col, $row, $buttons
 
 		$readonly = !empty($params['readonly']) ? ' readonly disabled' : '';
 
-		return '<div class="js-editor-none">'
+		HTMLHelper::_('webcomponent',
+			['joomla-editor-none' => 'system/webcomponents/joomla-editor-none.min.js'],
```
Comment (laoneo) [inline plugins/editors/none/none.php]: Guess this needs to be adapted?

Comment (anuragteapot) [reply plugins/editors/none/none.php]: @dgt41 can you fix this ?

**Comment by anuragteapot** [comment]
I have tested this item :white_check_mark: successfully on b6c6486cb93347e01d4f1862ab59144d0b4fc59b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19909">issues.joomla.org/tracker/joomla-cms/19909</a>.</sub>

**Comment by ghost** [comment]
@Samgithub1997 can you please retest?

**Comment by samarthsharma36** [comment]
I've exam in two hours. I'll test it by today

**Comment by laoneo** [comment]
If you can fix the conflict, then we are good to go here.

**Comment by samarthsharma36** [comment]
I have tested this item :white_check_mark: successfully on 2e1fcb4a7d0215e6b9d9134a12200845e5aa9879<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19909">issues.joomla.org/tracker/joomla-cms/19909</a>.</sub>

**Comment by samarthsharma36** [comment]
@laoneo , what conflict are you talking about ?

**Comment by laoneo** [comment]
There was a conflict, so I couldn't merge it. Dimitris fixed it.

**Pull request merged**